### PR TITLE
Add "deletedAt" property to all sync models

### PR DESCRIPTION
**Pull request by vinaysshenoy**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/overdue/communication/CommunicationPullResponse.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ data class CommunicationPullResponse(
     @Json(name = "communications")
     override val payloads: List<CommunicationPayload>,
 
-    @Json(name = "processed_since")
+    @Json(name = "process_token")
     override val processedSinceTimestamp: String
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/overdue/communication/CommunicationPullResponse.kt]: We could rename the variable `processedSinceTimestamp` in all our Response models.
Also in `DataPullResponse`.

File: app/src/main/java/org/simple/clinic/facility/FacilityPullResponse.kt [hunk 2]
```diff
@@ -10,6 +10,6 @@ data class FacilityPullResponse(
     @Json(name = "facilities")
     override val payloads: List<FacilityPayload>,
 
-    @Json(name = "processed_since")
+    @Json(name = "process_token")
     override val processedSinceTimestamp: String
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/facility/FacilityPullResponse.kt]: Rename this  `processedSinceTimestamp`-> `processToken`

File: app/src/main/java/org/simple/clinic/overdue/communication/CommunicationPullResponse.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/overdue/communication/CommunicationPullResponse.kt]: Makes sense, I'll do this.

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/overdue/communication/CommunicationPullResponse.kt]: Done.

**Pull request merged**